rust/connlib/snownet/tests/lib.rs

```diff
@@ -36,6 +37,53 @@ fn answer_after_stale_connection_does_not_panic() {
     alice.accept_answer(1, bob.public_key(), answer);
 }
 
+#[test]
+fn only_generate_candidate_event_after_answer() {
+    let local_candidate = SocketAddr::new(IpAddr::from(Ipv4Addr::LOCALHOST), 10000);
+
+    let mut alice = ClientNode::<u64>::new(
+        StaticSecret::random_from_rng(rand::thread_rng()),
+        Instant::now(),
+    );
+
+    alice.add_local_host_candidate(local_candidate).unwrap();
+
+    let mut bob = ServerNode::<u64>::new(
+        StaticSecret::random_from_rng(rand::thread_rng()),
+        Instant::now(),
+    );
+
+    let offer = alice.new_connection(1, HashSet::new(), HashSet::new());
+
+    while let Some(event) = alice.poll_event() {
+        assert!(!matches!(
+            event,
+            Event::SignalIceCandidate {
+                connection: _,
+                candidate: _
+            }
+        ));
+    }
```

Comment:
Which events do we generate upon `new_connection`?

Reply:
Can we not just assert `poll_event().is_none()`?

Reply:
Just wanted to assume as little as possible about the API

Reply:
If you want to keep asserting for the candidate, I'd suggest at least de-duplicating it between both so that it is clear that we are negating it in one case.